doc/api/fs.md

```diff
@@ -1101,11 +1101,68 @@ changes:
 * Returns: {AsyncIterator} An AsyncIterator that yields the paths of files
   that match the pattern.
 
+Retrieves the files matching the specified pattern(s).
+
+#### Pattern Syntax
+
+The glob implementation supports the following pattern syntax:
+
+**Basic Wildcards:**
+* `*` - Matches any number of characters within a single path segment
+* `?` - Matches exactly one character
+* `[abc]` - Matches any character in the brackets
+* `[a-z]` - Matches any character in the range
+* `[!abc]` or `[^abc]` - Matches any character not in the brackets
+* `[!a-z]` or `[^a-z]` - Matches any character not in the range
+
+**Globstar:**
+* `**` - Matches zero or more directories recursively
+* `**/*.js` - Matches all `.js` files in any subdirectory
+
+**Brace Expansion:**
+* `{a,b,c}` - Matches any of the comma-separated patterns
+* `{1..5}` - Matches any number in the range (1, 2, 3, 4, 5)
+* `*.{js,ts}` - Matches files with `.js` or `.ts` extensions
+
+**Extended Glob Patterns:**
+* `+(pattern)` - Matches one or more occurrences of the pattern
+* `*(pattern)` - Matches zero or more occurrences of the pattern
+* `?(pattern)` - Matches zero or one occurrence of the pattern
+* `!(pattern)` - Matches anything except the pattern
+
+#### Examples
+
 ```mjs
```

Comment:
Instead of one large code block with comments, I think it's better to split it out in multiple code snippets with documentation that is written out.